packages/browser/src/client.ts

```diff
@@ -40,7 +47,7 @@ export class BrowserClient extends BaseClient<BrowserOptions> {
    *
    * @param options Configuration options for this SDK.
    */
-  public constructor(options: BrowserOptions = {}) {
+  public constructor(options: BrowserOptions = {}, transport: Transport, newTransport?: NewTransport) {
```

Comment:
instead of it being a top level property, what do you think about making it part of options?

This might mean we need to introduce two sets of options for the SDK. There is the options that `SDK.init` expects, and options that the client constructor expects.

Me and @timfish discussed this previously. It looks someting like this:

```ts
interface Options {
  dsn: string,
  // optional property here
  transport?: NewTransport
}

interface ClientOptions {
  dsn: DsnComponents
  // required for client constructor
  // logic that creates a class will choose an appropriate transport
  transport: NewTransport
}
```

This may be easier to just do when we are looking at the transports in general, let's chat about it during daily.

Reply:
I think this is a good idea because it also might clear up confusion around where to specify the (custom) transport.

So if I understand this idea correctly, the `ClientOptions` would always contain a valid `NewTransport` (and `Transport` for the time being)? So if there is no `transport` specified in `Options`, the logic creating the `ClientOptions` would fall back to our default ones?

Reply:
Also, what about `DsnComponents` in the `ClientOptions` example?

Reply:
> So if I understand this idea correctly, the ClientOptions would always contain a valid NewTransport (and Transport for the time being)? So if there is no transport specified in Options, the logic creating the ClientOptions would fall back to our default ones?

Yes exactly.

> Also, what about DsnComponents in the ClientOptions example?

This is how we map a string dsn to it's various components. It should actually be optional actually. Just added as an another example of options that need to be mapped. See https://github.com/getsentry/sentry-javascript/blob/4e722eb8778e27d7910e96ccb1aac108bcbea146/packages/core/src/baseclient.ts#L81

Reply:
Just to update GH: We decided to go with the constructor arguments in this PR and change them in a later PR to become properties in `options`. This will be done once we settled on the user-facing and client-specific `Options` discussion.